plugins/org.yakindu.sct.model.sgraph/src/org/yakindu/sct/model/sgraph/resource/AbstractSCTResource.java

```diff
@@ -343,6 +343,16 @@ public void linkSpecificationElements() {
 			}
 		}
 	}
+	
+	public void parseSpecificationElements() {
+		TreeIterator<EObject> iter = getAllContents();
+		while (iter.hasNext()) {
+			EObject currentObject = iter.next();
+			if (currentObject instanceof SpecificationElement) {
+				parseSpecificationElement((SpecificationElement) currentObject);
```

Comment:
can we add an iter.prune() here?

Reply:
If a SpecificationElement is a CompositeState, the State can contain further States and Transitions. If a refactoring made changes to those elements then they would't be parsed with iter.prune(), won't they?

Reply:
yes you are right